transformer_engine/common/CMakeLists.txt

```diff
@@ -141,6 +149,25 @@ if (NVTE_ENABLE_NVSHMEM)
     target_include_directories(transformer_engine PUBLIC ${NVSHMEMAPI_INCLUDE_DIR})
 endif()
 
+option(NVTE_WITH_CUBLASMP "Use cuBLASMp for tensor parallel GEMMs" OFF)
+if (NVTE_WITH_CUBLASMP)
+    target_compile_definitions(transformer_engine PRIVATE NVTE_WITH_CUBLASMP)
+    target_include_directories(transformer_engine PRIVATE ${CUBLASMP_DIR}/include ${NVSHMEM_DIR}/include)
+    find_library(CUBLASMP_LIB
+                 NAMES cublasmp libcublasmp
+                 PATHS ${CUBLASMP_DIR}
+                 PATH_SUFFIXES lib
+                 REQUIRED)
+    find_library(NVSHMEM_HOST_LIB
+                 NAMES nvshmem_host libnvshmem_host.so.3
+                 PATHS ${NVSHMEM_DIR}
+                 PATH_SUFFIXES lib
+                 REQUIRED)
+  target_link_libraries(transformer_engine PUBLIC ${CUBLASMP_LIB} ${NVSHMEM_HOST_LIB})
```

Comment:
Do we see linker errors when importing `libtransformer_engine.so`? If we do, we might need to manually import cuBLASMp and NVSHMEM like we do with other NVIDIA libs:
https://github.com/NVIDIA/TransformerEngine/blob/92f431bfee3bebbf49d1cc6f6bc37796bffd8bb7/transformer_engine/common/__init__.py#L304-L308
This is because the NVIDIA libs might be installed in places where the linker can't automatically find them, especially if users install them via Pip.